homeassistant/components/guardian/__init__.py

```diff
@@ -429,23 +431,73 @@ def __init__(  # pylint: disable=super-init-not-called
         """Initialize."""
         self._attr_extra_state_attributes = {}
         self._entry = entry
+        self._rebooting = False
+        self._signal_reboot_completed = SIGNAL_REBOOT_COMPLETED.format(entry.entry_id)
+        self._signal_reboot_requested = SIGNAL_REBOOT_REQUESTED.format(entry.entry_id)
         self.entity_description = description
 
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available.
+
+        Since we don't call our parent's __init__(), we remake our own version of this
+        property so that we can force entity available when desired (e.g., a reboot).
+        """
+        return self._attr_available
+
+    @callback
+    def _async_handle_coordinator_update(self) -> None:
+        """Respond to a GuardianDataUpdateCoordinator update."""
+        # If *any* coordinator successfully updates while the entity is marked as
+        # "rebooting," the API is back up and all entities (even those with different
+        # coordinators) should become available:
+        if self._rebooting:
+            async_dispatcher_send(self.hass, self._signal_reboot_completed)
+            return
+
+        self._async_update_from_latest_data()
+        self.async_write_ha_state()
+
     @callback
     def _async_update_from_latest_data(self) -> None:
-        """Update the entity.
+        """Update the entity's underlying data.
 
         This should be extended by Guardian platforms.
         """
 
+    async def async_added_to_hass(self) -> None:
+        """Perform tasks when the entity is added."""
+        self._async_update_from_latest_data()
+        self.async_write_ha_state()
+
+        @callback
+        def async_reboot_completed() -> None:
+            """Respond to a completed controller reboot."""
+            self._attr_available = True
+            self._rebooting = False
+            self._async_handle_coordinator_update()
+
+        @callback
+        def async_reboot_requested() -> None:
+            """Respond to a requested controller reboot."""
+            self._attr_available = False
+            self._rebooting = True
```

Comment:
Instead of this, why not use `self.coordinator.async_set_update_error`. Maybe extend that method to allow not logging the error.

Reply:
As discussed, one challenge lies with the fact that we make use of multiple coordinators. I see how the current PR is duplicative; that said, one nice thing is that when one coordinator recovers, _all_ entities get marked as `available` (I infer that if one API is back, the whole thing is back).

If we rely on each coordinator to handle the availability of its entities, we could face a scenario where entities become available in a disjointed manner after a reboot.

None of my immediate solutions are pleasing. I'll keep thinking about it.

Reply:
I thought on it, and I haven't found a way to accomplish both outcomes (removal of duplicative work _and_ recovery of all entities):

1. Use coordinator callbacks to mark `last_update_success` on every coordinator as `True` – that feels invasive; why should "outsiders" touch that attribute?
2. Use coordinator callbacks to send new API requests – this really only speeds up the recovery (and does so with tons of different, overlapping API requests).

Reply:
@balloob Latest push removes the redundancy while retaining the ability to update all entities at once. I elected to go for simpler methods on my coordinator subclass. Let me know?

Reply:
Okay, I put more of the logic directly in the coordinator... That said, the dispatcher still seems to be the best way to ensure all coordinators are simultaneously aware of the reboot request.

We get lucky here because all coordinators (a) go unavailable at the same moment and (b) have the same update interval (and so will all go available at roughly the same moment). In #75227, RainMachine's coordinators have different intervals...will need to get creative there.